scrapy/spidermiddlewares/offsite.py

```diff
@@ -28,7 +28,7 @@ def from_crawler(cls, crawler):
     def process_spider_output(self, response, result, spider):
         for x in result:
             if isinstance(x, Request):
-                if x.dont_filter or self.should_follow(x, spider):
+                if x.meta.get('allow_offsite_requests', False) or self.should_follow(x, spider):
```

Comment:
I am thinking this would keep this change backward-compatible:
```suggestion
                if x.meta.get('allow_offsite_requests', x.dont_filter) or self.should_follow(x, spider):
```

If you agree, then I suggest you also change your test changes, so that you keep existing tests unchanged and simply add additional tests scenarios.

Reply:
I'm not too sure about this even with the advantage of backward compatibility because when we want to duplicate a request but not skip the offsite filter ( which I assume is the case for majority of instances where we set dont_filter=True) we have to add meta={'allow_offsite': False} which is extra work. Plus there will be two ways to skip offsite filtering.

Hence I'm proposing to do this in a backward incompatible way if possible.